src/transformers/pipelines/audio_classification.py

```diff
@@ -126,6 +114,16 @@ def __call__(
             - **label** (:obj:`str`) -- The label predicted.
             - **score** (:obj:`float`) -- The corresponding probability.
         """
+        return super().__call__(inputs, **kwargs)
+
+    def set_parameters(self, top_k=None, **kwargs):
+        # No parameters on this pipeline right now
+        if top_k is not None:
+            self.top_k = top_k
+        if self.top_k > self.model.config.num_labels:
+            self.top_k = self.model.config.num_labels
```

Comment:
The only issue I'm seeing with this method is that it has side effects on the pipeline state, which is surprising in some cases (e.g., on `__call__`).

For example here:
```py
>>> pp = pipeline("text-generation", max_length=2)
```
You would expect all generations coming from this pipeline to have a `max_length` of 2. And this works well.

When using the same pipeline for inference, with a `max_length` override, this too works as one would expect:
```py
>>> pp(["ok", "nice"], max_length=8)
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
[[{'generated_text': 'ok\n\nAce is my friend'}], [{'generated_text': "nice will always tell you, 'I"}]
```

What is unexpected is that now the default max length changed to 8; when this should IMO be an override and not set:

```py
>>> pp(["ok", "nice"])
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
Setting `pad_token_id` to `eos_token_id`:50256 for open-end generation.
[[{'generated_text': 'ok and Aarhus in Copenhagen.'}], [{'generated_text': 'nice for a while at first. I'}]
```

Reply:
You are entirely right ! I think we need to change that, and hopefully this will also help improve the cleanliness there.

Thanks I think pointing that out gave me an idea on how to dramatically improve this.